app/src/utils/utils.js

```diff
@@ -163,26 +163,35 @@ module.exports = {
    */
   findObject: function(keyPairs, objects) {
     var filteredObjects = _.filter(objects, function(o) {
-      var found = true;
-
+      var found = false;
+      var results = [];
       Object.keys(keyPairs).forEach(function(keyName) {
         var keyVal = keyPairs[keyName];
         if (_.isEmpty(keyVal)) {
-          found = false;
+          results.push(false);
           log.error(
             "",
             "An empty search key was provided preventing an object match."
           );
         }
-        found &= o[keyName] == keyVal;
+        results.push(o[keyName] == keyVal);
+      })
+      results.forEach(function(item) {
+        if (results[0] != item) {
+          throw new Error(
+            "Illegal state: provided search keys were partially matched on the target objects"
+          );
+        }
+        found = item;
```

Comment:
This is a version using [`Array.every()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/every), note how `found` is only initialized when `results` is filled up:
```javascript
var filteredObjects = _.filter(objects, function(o) {
  var results = [];
  Object.keys(keyPairs).forEach(function(keyName) {
    var keyVal = keyPairs[keyName];
    if (_.isEmpty(keyVal)) {
      results.push(false);
      log.error(
        "",
        "An empty search key was provided preventing an object match."
      );
    }
    results.push(o[keyName] == keyVal);
  })

  var found = results[0];
  if (!results.every( val => val === found )) {
    throw new Error(
      "Illegal state: provided search keys were partially matched on the target objects"
    );
  }
  return found;
});
```